src/xdist/dsession.py

```diff
@@ -372,32 +379,51 @@ def handle_crashitem(self, nodeid, worker):
         self.config.hook.pytest_runtest_logreport(report=rep)
 
 
+class WorkerStatus(Enum):
+    """Status of each worker during creation/collection."""
+
+    # Worker spec has just been created.
+    Created = auto()
+
+    # Worker has been initialized.
+    Initialized = auto()
+
+    # Worker is now ready for collection.
+    Ready = auto()
```

Comment:
If you can bare the length, would name it `ReadyForCollection`, because `Ready` is somewhat ambiguous (I would have thought that the "collection done" one is "ready").

Reply:
Done, thanks!